db/migrate/20160712044409_create_join_table_storage_profile_storage.rb

```diff
@@ -0,0 +1,10 @@
+class CreateJoinTableStorageProfileStorage < ActiveRecord::Migration[5.0]
+  def change
+    create_table :storage_profile_storages do |t|
+      t.bigint  :storage_profile_id
+      t.bigint  :storage_id
+      t.index [:storage_id, :storage_profile_id], :name => 'storage_to_profile'
+      t.index [:storage_profile_id, :storage_id], :name => 'profile_to_storage'
```

Comment:
Just off a paired session with @Fryguy to compare the effects of using composite/simple index.  The short answer is we'll keep 2 simple index on ```storage_id``` and ```storage_profile_id``` for lookups from both direction.
For details, can refer to 
* compound index: https://explain.depesz.com/s/NkT
* simple index: https://explain.depesz.com/s/Dix